composer.json

```diff
@@ -71,15 +71,16 @@
         "php": ">=5.4.0",
         "ext-mbstring": "*",
         "ext-ctype": "*",
-        "lib-pcre": "*",
+        "ext-pcre": "*",
+        "ext-openssl": "*",
         "yiisoft/yii2-composer": "~2.0.4",
-        "ezyang/htmlpurifier": "~4.6",
+        "ezyang/htmlpurifier": "~4.14",
         "cebe/markdown": "~1.0.0 | ~1.1.0 | ~1.2.0",
-        "bower-asset/jquery": "3.6.*@stable | 3.5.*@stable | 3.4.*@stable | 3.3.*@stable | 3.2.*@stable | 3.1.*@stable | 2.2.*@stable | 2.1.*@stable | 1.11.*@stable | 1.12.*@stable",
```

Comment:
I can set `'@bower' => '@npm'`(here) or copy `@npm` packages to `@bower`(in `yiisoft/yii2-composer`)

Reply:
> That would be a breaking change because it will write assets to different directory. It is overwritten in app templates but there could be older apps around.

Shouldn't that only be handled in asset-bundles, so I would not really matter where the files are located?
It would only affect an application, when you do eg. some bundling manually.

But yes, this is somehow only a technical view, in real-world applications you might run into problems.

What's more of a concern to me is that usually npm packages load much more source-code than bower.

Reply:
@schmunk42 bower is abandoned less year ago
> ...psst! While Bower is maintained, we recommend using [Yarn](https://yarnpkg.com/) and [Webpack](https://webpack.js.org/) or [Parcel](https://parceljs.org/) for front-end projects [read how to migrate](https://bower.io/blog/2017/how-to-migrate-away-from-bower/)

Reply:
I was talking about `devDependencies`, see example:

```
root@804503013acb:/app# composer require bower-asset/selectize.js
Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9005 (through xdebug.client_host/xdebug.client_port) :-(
Using version ^0.12.6 for bower-asset/selectize.js
./composer.json has been updated
Running composer update bower-asset/selectize.js
Loading composer repositories with package information
Updating dependencies
Lock file operations: 4 installs, 0 updates, 0 removals
  - Locking bower-asset/jquery (3.6.0)
  - Locking bower-asset/microplugin (v0.0.3)
  - Locking bower-asset/selectize.js (v0.12.6)
  - Locking bower-asset/sifter (v0.5.4)
Writing lock file
Installing dependencies from lock file (including require-dev)



root@804503013acb:/app# composer require npm-asset/selectize.js
Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9005 (through xdebug.client_host/xdebug.client_port) :-(
Using version ^0.13.3 for npm-asset/selectize.js
./composer.json has been updated
Running composer update npm-asset/selectize.js
Loading composer repositories with package information
Updating dependencies
Lock file operations: 14 installs, 0 updates, 0 removals
  - Locking npm-asset/ansicolors (0.2.1)
  - Locking npm-asset/async (2.6.3)
  - Locking npm-asset/cardinal (1.0.0)
  - Locking npm-asset/csv-parse (4.12.0)
  - Locking npm-asset/esprima (3.0.0)
  - Locking npm-asset/humanize (0.0.9)
  - Locking npm-asset/lodash (4.17.21-patch1)
  - Locking npm-asset/microplugin (0.0.3)
  - Locking npm-asset/minimist (0.0.10)
  - Locking npm-asset/optimist (0.6.1)
  - Locking npm-asset/redeyed (1.0.1)
  - Locking npm-asset/selectize.js (v0.13.3)
  - Locking npm-asset/sifter (0.5.4)
  - Locking npm-asset/wordwrap (0.0.3)
Writing lock file
Installing dependencies from lock file (including require-dev)
```

The npm version does not even install jQuery, which is actually required, but not installed since it is listed as `peerDependency` https://github.com/selectize/selectize.js/blob/master/package.json#L69-L71

While **bower** is deprecated, its design still fits much better the usage with composer - while I do not like the fact, it's still a reality.

Reply:
@schmunk42 we not develop that's packages, we use it "as is" - size of `devDependencies` not matter, we use production versions and that's versions same:

![изображение](https://user-images.githubusercontent.com/3521094/171012752-14015360-f1c1-4553-a10d-d22da9478828.png)

Reply:
I am NOT talking about the single package size from asset-packagist, but the dependencies installed when using bower or npm.

Also `--no-dev` makes no difference:

```
root@804503013acb:/app# composer update --no-dev
Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9005 (through xdebug.client_host/xdebug.client_port) :-(
Loading composer repositories with package information
Updating dependencies
Nothing to modify in lock file
Installing dependencies from lock file
Package operations: 18 installs, 0 updates, 0 removals
  - Installing bower-asset/sifter (v0.5.4): Extracting archive
  - Installing bower-asset/microplugin (v0.0.3): Extracting archive
  - Installing bower-asset/jquery (3.6.0): Extracting archive
  - Installing bower-asset/selectize.js (0.12.6-patch1): Extracting archive
  - Installing npm-asset/ansicolors (0.2.1): Extracting archive
  - Installing npm-asset/lodash (4.17.21-patch1): Extracting archive
  - Installing npm-asset/minimist (0.0.10): Extracting archive
  - Installing npm-asset/esprima (3.0.0): Extracting archive
  - Installing npm-asset/redeyed (1.0.1): Extracting archive
  - Installing npm-asset/wordwrap (0.0.3): Extracting archive
  - Installing npm-asset/optimist (0.6.1): Extracting archive
  - Installing npm-asset/humanize (0.0.9): Extracting archive
  - Installing npm-asset/csv-parse (4.12.0): Extracting archive
  - Installing npm-asset/cardinal (1.0.0): Extracting archive
  - Installing npm-asset/async (2.6.3): Extracting archive
  - Installing npm-asset/sifter (0.5.4): Extracting archive
  - Installing npm-asset/microplugin (0.0.3): Extracting archive
  - Installing npm-asset/selectize.js (v0.13.3): Extracting archive
```

See https://github.com/fxpio/composer-asset-plugin/issues/269#issuecomment-371810005 for an explanation.
It has already been improved, but the issues mentioned still persist.